src/pages/AgencyContractsPage/Employment.tsx

```diff
@@ -34,16 +35,22 @@ const EmploymentPage: React.FC<EmploymentProps> = () => {
   const currentForm: any = useSelector((state: any) => state.employmentAgreements.currentAgreement)
   const agreements = useSelector((state: any) => state.employmentAgreements.agreements) || []
   const { data, ...user } = useSelector((state: any) => state.user)
-  const userContacts: User[] = useSelector((state: any) => state.user.contacts)
+  const userContacts = useSelector((state: IRootState) => state.businessContracts.contracts)
   const { t } = useTranslation()
 
-  userContacts.forEach((user) => {
-    if (user.userType == 'worker') {
-      workers.push(user)
-    } else if (user.userType == 'business') {
-      businesses.push(user)
-    }
-  })
+  if (userContacts && userContacts.length > 0) {
+    userContacts.forEach((user: any) => {
+      if (user.target.userType == 'worker') {
+        workers.push(user.target)
+      } else if (user.target.userType == 'business') {
+        businesses.push(user.target)
```

Comment:
```suggestion
    userContracts.forEach((contract: any) => {
      if (contract.target.userType == 'worker') {
        workers.push(contract.target)
      } else if (contract.target.userType == 'business') {
        businesses.push(contract.target)
```
Contract on varmaan kuvaavampi?